dask/blockwise.py

```diff
@@ -531,28 +532,7 @@ def __dask_distributed_pack__(
             }
             inline_tasks = inline_tasks or blockwise_dep.produces_tasks
 
-        # Dump (pickle + cache) the function here if we know `make_blockwise_graph`
-        # will NOT be producing "nested" tasks (via `__dask_distributed_unpack__`).
-        #
-        # If `make_blockwise_graph` DOES need to produce nested tasks later on, it
-        # will need to call `to_serialize` on the entire task.  That will be a
-        # problem if the function was already pickled here. Therefore, we want to
-        # call `to_serialize` on the function if we know there will be nested tasks.
-        #
-        # We know there will be nested tasks if either:
-        #   (1) `concatenate=True`   # Check `self.concatenate`
-        #   (2) `inline_tasks=True`  # Check `BlockwiseDep.produces_tasks`
-        #
-        # We do not call `to_serialize` in ALL cases, because that code path does
-        # not cache the function on the scheduler or worker (or warn if there are
-        # large objects being passed into the graph).  However, in the future,
-        # single-pass serialization improvements should allow us to remove this
-        # special logic altogether.
-        func = (
-            to_serialize(dsk[0])
-            if (self.concatenate or inline_tasks)
-            else dumps_function(dsk[0])
-        )
+        func = PickledCallable.serialize(dsk[0])
```

Comment:
This is nice, and it reminds me that we should probably abandon the use of [CallableLazyImport](https://github.com/dask/dask/blob/965e35fe5a9423f550991bde08116a4984498e68/dask/layers.py#L23) in existing and future `Layer` definitions.